### PR TITLE
[Docs] Introduce Performance article

### PR DESCRIPTION
**Pull request by martintmk**
### Details on the issue fix or feature implementation

Contributes to #1091 

### Confirm the following

- [x]  I started this PR by branching from the head of the default branch
- [x]  I have targeted the PR to merge into the default branch
- [x]  I have included unit tests for the issue/feature
- [x]  I have successfully run a local build

### CODE REVIEW

File: docs/advanced/performance.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+// This approach uses a static lambda, avoiding allocations.
+// The "userId" is stored as state, and the lambda reads it.
+await resiliencePipeline.ExecuteAsync(
+    static (state, cancellationToken) => GetMemberAsync(state, cancellationToken),
+    userId,
+    cancellationToken);
+```
+<!-- endSnippet -->
+
+### Use switch expressions for predicates
```
Comment (adamnova) [inline docs/advanced/performance.md:48]: I would maybe mention 'why', to explain why is it faster. Similarly to the use of static lambdas. Small plus in that could also be knowledge sharing, they might use the same pattern somewhere else and realise there is a better alternative.

**Review by adamnova — APPROVED** [review]
Looks good 👍

File: docs/advanced/performance.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+| Method              |     Mean |     Error |    StdDev | Ratio | RatioSD |   Gen0 | Allocated | Alloc Ratio |
+| ------------------- | -------: | --------: | --------: | ----: | ------: | -----: | --------: | ----------: |
+| Execute policy v7   | 2.277 μs | 0.0133 μs | 0.0191 μs |  1.00 |    0.00 | 0.1106 |    2824 B |        1.00 |
+| Execute pipeline v8 | 2.089 μs | 0.0105 μs | 0.0157 μs |  0.92 |    0.01 |      - |      40 B |        0.01 |
+
+---
+
+Compared to older versions, Polly v8 is both faster and more memory efficient.
+
+## Performance tips
```
Comment (juraj-blazek) [inline docs/advanced/performance.md:24]: Shall we also add a tip to use the `Outcome` based API to avoid exceptions?

Comment (martintmk) [reply docs/advanced/performance.md:24]: True, will do

**Pull request merged**